src/main/java/io/supertokens/webserver/api/dashboard/DashboardUserAPI.java

```diff
@@ -0,0 +1,207 @@
+                response.addProperty("status", "PASSWORD_WEAK_ERROR");
+                super.sendJsonResponse(200, response, resp);
+                return;
+            }
+
+            Dashboard.signUpDashboardUser(main, email, password);
+
+            JsonObject response = new JsonObject();
+            response.addProperty("status", "OK");
+            super.sendJsonResponse(200, response, resp);
+
+        } catch (DuplicateEmailException e) {
+            JsonObject response = new JsonObject();
+            response.addProperty("status", "EMAIL_ALREADY_EXISTS_ERROR");
+            super.sendJsonResponse(200, response, resp);
+        } catch (StorageQueryException e) {
+            throw new ServletException(e);
+        }
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+
+        String newEmail = InputParser.parseStringOrThrowError(input, "newEmail", true);
+        if (newEmail != null) {
+            // normalize new email
+            newEmail = normalizeStringParam(newEmail, "newEmail");
+
+            // check if the newEmail is in valid format
+            if (!Dashboard.isValidEmail(newEmail)) {
+                JsonObject response = new JsonObject();
+                response.addProperty("status", "INVALID_EMAIL_ERROR");
+                super.sendJsonResponse(200, response, resp);
+                return;
+            }
+        }
+
+        String newPassword = InputParser.parseStringOrThrowError(input, "newPassword", true);
+        if (newPassword != null) {
+            // normalize new password
+            newPassword = normalizeStringParam(newPassword, "newPassword");
+            // check if the new password is strong
+            if (Dashboard.isStrongPassword(newPassword)) {
+                JsonObject response = new JsonObject();
+                response.addProperty("status", "PASSWORD_WEAK_ERROR");
+                super.sendJsonResponse(200, response, resp);
+                return;
+            }
+        }
+
+        try {
+            String userId = InputParser.parseStringOrThrowError(input, "userId", true);
+            if (userId != null) {
+                // normalize userId
+                userId = normalizeStringParam(userId, "userId");
+
+                Dashboard.updateUsersCredentialsWithUserId(main, userId, newEmail, newPassword);
+                JsonObject response = new JsonObject();
+
+                response.addProperty("status", "OK");
+                super.sendJsonResponse(200, response, resp);
+            }
+
+            String email = InputParser.parseStringOrThrowError(input, "email", true);
+            if (email != null) {
+                // normalize userId
+                email = normalizeStringParam(email, "email");
+                Dashboard.updateUsersCredentialsWithEmail(main, email, newEmail, newPassword);
```

Comment:
Change this to get the user id using the email and then call the method that takes user id as an input